### PR TITLE
Add typings 

### PR DESCRIPTION
**Pull request by Leeleo3x**
Add typings for: ContractDefinition, StateVariableDeclaration, FunctionDefinition, ParameterList, VariableDeclaration, ArrayTypeName, Mapping, ForStatement, ElementaryTypeName, FunctionCall, ElementaryTypeNameExpression.

This helps issue #48

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -157,9 +159,14 @@ export interface ModifierInvocation extends BaseASTNode {
 }
 export interface FunctionDefinition extends BaseASTNode {
   type: 'FunctionDefinition';
-  name: string;
+  name?: string;
```
Comment (yxliang01) [inline index.d.ts]: @Leeleo3x Can you give an example when `name` is undefined?

File: index.d.ts [hunk 2]
```diff
@@ -157,9 +159,14 @@ export interface ModifierInvocation extends BaseASTNode {
...
   parameters: ParameterList;
-  body: Block | null;
+  modifiers: ModifierInvocation[];
+  stateMutability?: 'pure' | 'constant' | 'payable' | 'view'
+  visibility: 'default' | 'external' | 'internal' | 'public' | 'private';
+  isConstructor: boolean;
+  returnParameters?: ParameterList;
+  body?: Block;
```
Comment (yxliang01) [inline index.d.ts]: @Leeleo3x Can you give an example when `body` is undefined?

File: index.d.ts (hunk 1 above)
Comment (Leeleo3x) [reply index.d.ts]:
```
constructor () {}
```
or
```
function () payable {} 
```

File: index.d.ts (hunk 2 above)
Comment (Leeleo3x) [reply index.d.ts]:
```
contract ERC20Interface {

    function totalSupply() public view returns (uint);

    function balanceOf(address tokenOwner) public view returns (uint balance);

    function allowance(address tokenOwner, address spender) public view returns (uint remaining);

    function transfer(address to, uint tokens) public returns (bool success);

    function approve(address spender, uint tokens) public returns (bool success);

    function transferFrom(address from, address to, uint tokens) public returns (bool success);


    event Transfer(address indexed from, address indexed to, uint tokens);

    event Approval(address indexed tokenOwner, address indexed spender, uint tokens);

}
```

File: index.d.ts (hunk 1 above)
Comment (yxliang01) [reply index.d.ts]: @Leeleo3x For fallback (default) function, it has value of `""` (empty string). I will try to see constructor later.

Comment (Leeleo3x) [reply index.d.ts]: You may check the code

https://github.com/federicobond/solidity-parser-antlr/blob/master/src/ASTBuilder.js#L168

**Comment by federicobond** [comment]
Thank you @Leeleo3x and @yxliang01! I have pushed release 0.4.3 to `npm` with the latest changes.

**Pull request merged**